packages/space-opera/src/components/model_viewer_snippet/components/parsing.ts

```diff
@@ -0,0 +1,133 @@
+  'shadow-softness',
+  'max-camera-orbit',
+  'min-camera-orbit',
+  'max-field-of-view',
+  'min-field-of-view',
+  'autoplay',
+  'animation-name',
+  'camera-orbit',
+  'camera-target',
+  'field-of-view',
+  'ar',
+  'ar-modes',
+  'ios-src'
+];
+
+// Creating object in this way such that "spread" works.
+// https://open-wc.org/docs/development/lit-helpers/#regular-spread
+export function parseExtraAttributes(snippet: string): string {
+  // Parse snippet and extract attributes from model-viewer
+  const parsedInput = new DOMParser().parseFromString(snippet, 'text/html');
+  const modelViewer = parsedInput.body.getElementsByTagName('model-viewer')[0];
+  let extraAttributes: any = {};
+  const attributes = modelViewer.attributes;
+
+  // Loop through every attribute, only add the attributes to extraAttributes
+  // if they are not a part of config or arConfig
+  for (let i = 0; i < attributes.length; i++) {
+    const name: string = attributes[i].name;
+    const value = attributes[i].value;
+    // if neither config has the key, add it to extra attribute string
+    if (!(ATTRIBUTE_LIST.includes(name))) {
+      if (value.length === 0) {
+        // Boolean attribute is true
+        extraAttributes[`?${name}`] = true;
+      } else {
+        // Normal attribute is set to it's snippet's value
+        extraAttributes[`${name}`] = value;
+      }
+    }
+  }
+  return extraAttributes;
+}
+
+// logic similar to parseSnippet inside of editing adapter.
+export function parseSnippetAr(snippet: string): ArConfigState {
+  const parsedInput = new DOMParser().parseFromString(snippet, 'text/html');
+  const modelViewer = parsedInput.body.getElementsByTagName('model-viewer')[0];
+  const arConfig: ArConfigState = {};
+  arConfig.ar = modelViewer.hasAttribute('ar') || undefined;
+  arConfig.arModes = modelViewer.getAttribute('ar-modes') || undefined;
+  arConfig.iosSrc = modelViewer.getAttribute('ios-src') || undefined;
+  return arConfig
+}
+
+function tryParseNumberAttribute(element: Element, attribute: string): number|
+    undefined {
+  const attributeValue = element.getAttribute(attribute);
+  if (element.hasAttribute(attribute) && attributeValue !== '' &&
+      isFinite(Number(attributeValue))) {
+    return Number(attributeValue);
+  }
+  return undefined;
+}
+
+/**
+ * Parse a string representation of a model-viewer tag.
+ */
+export function parseSnippet(snippet: string): ModelViewerConfig {
+  const parsedInput = new DOMParser().parseFromString(snippet, 'text/html');
+  const modelViewer = parsedInput.body.getElementsByTagName('model-viewer')[0];
+  const config: ModelViewerConfig = {};
+  config.src = modelViewer.getAttribute('src') || undefined;
+  config.autoRotate = modelViewer.hasAttribute('auto-rotate');
+  config.cameraControls = modelViewer.hasAttribute('camera-controls');
+  // NOTE: bgcolor not well-supported yet, since real mv tags put this in
+  // the style tag. Will need to reconsider how we approach style in general.
```

Comment:
I'm guessing this is an old comment? Do we actually use bgcolor anywhere?

Reply:
you're right, this is an old comment